pkg/brokerapi/openservicebroker/open_service_broker_client.go

```diff
@@ -78,59 +116,78 @@ func (c *openServiceBrokerClient) GetCatalog() (*brokerapi.Catalog, error) {
 	return &catalog, nil
 }
 
-func (c *openServiceBrokerClient) CreateServiceInstance(ID string, req *brokerapi.ServiceInstanceRequest) (*brokerapi.ServiceInstance, error) {
-	jsonBytes, err := json.Marshal(req)
-	if err != nil {
-		return nil, err
-	}
-
+func (c *openServiceBrokerClient) CreateServiceInstance(ID string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.CreateServiceInstanceResponse, error) {
 	url := fmt.Sprintf(serviceInstanceFormatString, c.broker.Spec.URL, ID)
-
 	// TODO: Handle the auth
-	createHTTPReq, err := http.NewRequest("PUT", url, bytes.NewReader(jsonBytes))
+	resp, err := util.SendRequest(c.client, http.MethodPut, url, req)
 	if err != nil {
-		return nil, err
+		return nil, errRequest{message: err.Error()}
 	}
+	defer resp.Body.Close()
 
-	glog.Infof("Doing a request to: %s", url)
-	resp, err := c.client.Do(createHTTPReq)
-	if err != nil {
-		return nil, err
+	createServiceInstanceResponse := brokerapi.CreateServiceInstanceResponse{}
+	if err := util.ResponseBodyToObject(resp, &createServiceInstanceResponse); err != nil {
+		return nil, errResponse{message: err.Error()}
 	}
-	defer resp.Body.Close()
 
-	// TODO: Align this with the actual response model.
-	si := brokerapi.ServiceInstance{}
-	if err := util.ResponseBodyToObject(resp, &si); err != nil {
-		return nil, err
+	switch resp.StatusCode {
+	case http.StatusCreated:
+		return &createServiceInstanceResponse, nil
+	case http.StatusOK:
+		return &createServiceInstanceResponse, nil
+	case http.StatusAccepted:
+		glog.Infof("Asynchronous response received. Polling broker.")
+		if err := c.pollBroker(ID, createServiceInstanceResponse.Operation); err != nil {
+			return nil, err
+		}
+
+		return &createServiceInstanceResponse, nil
+	case http.StatusConflict:
+		return nil, errConflict
+	case http.StatusUnprocessableEntity:
+		return nil, errAsynchronous
+	default:
+		return nil, errStatusCode{statusCode: resp.StatusCode}
 	}
-	return &si, nil
 }
 
-func (c *openServiceBrokerClient) UpdateServiceInstance(ID string, req *brokerapi.ServiceInstanceRequest) (*brokerapi.ServiceInstance, error) {
+func (c *openServiceBrokerClient) UpdateServiceInstance(ID string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.ServiceInstance, error) {
 	// TODO: https://github.com/kubernetes-incubator/service-catalog/issues/114
 	return nil, fmt.Errorf("Not implemented")
 }
 
-func (c *openServiceBrokerClient) DeleteServiceInstance(ID string) error {
+func (c *openServiceBrokerClient) DeleteServiceInstance(ID string, req *brokerapi.DeleteServiceInstanceRequest) error {
```

Comment:
It's not entirely body-less. Deleting the service instance requires both the service ID and the plan ID. And we will need to also pass the `accepts_incomplete` flag presumably because this controller should be able to handle async calls.

We could have both the service ID and plan ID be passed in as arguments, and then add the accepts_incomplete automatically in the broker client. However, if that's the case it seems weird to me to have some operations accept a request object and others accept a list of attributes that will then be assembled into a request object, so I wanted to standardize a bit.